packages/cli/cloud/src/services/strapi-info-save.ts

```diff
@@ -1,20 +1,35 @@
 import fse from 'fs-extra';
 import path from 'path';
-import type { ProjectInfos } from './cli-api';
+import type { ProjectInfo } from './cli-api';
 
 export const LOCAL_SAVE_FILENAME = '.strapi-cloud.json';
 
 export type LocalSave = {
-  project?: Omit<ProjectInfos, 'id'>;
+  project?: Omit<ProjectInfo, 'id'>;
 };
 
-export async function save(data: LocalSave, { directoryPath }: { directoryPath?: string } = {}) {
+export async function addEnvironment(
+  targetEnvironment: string | undefined,
+  { directoryPath }: { directoryPath?: string } = {}
+) {
   const alreadyInFileData = await retrieve({ directoryPath });
-  const storedData = { ...alreadyInFileData, ...data };
+  if (!alreadyInFileData || !alreadyInFileData.project) {
+    throw new Error('No valid configuration found in the local configuration file.');
+  }
+  const dataToSave: LocalSave = {
+    ...alreadyInFileData,
+    project: {
+      ...alreadyInFileData.project,
+      targetEnvironment,
+    },
+  };
+  await save(dataToSave, { directoryPath });
+}
+
+export async function save(data: LocalSave, { directoryPath }: { directoryPath?: string } = {}) {
   const pathToFile = path.join(directoryPath || process.cwd(), LOCAL_SAVE_FILENAME);
-  // Ensure the directory exists
   await fse.ensureDir(path.dirname(pathToFile));
-  await fse.writeJson(pathToFile, storedData, { encoding: 'utf8' });
```

Comment:
Why did you removed the 
```
const storedData = { ...alreadyInFileData, ...data };
```

part?

Reply:
I think we should use `save` to store the configuration when linking a project initially. It’s safer to overwrite all config data as this avoids carrying over any outdated data from a previously linked project. Then use a separate function to add specific details as needed—as I do with addEnvironment. Since addEnvironment is our only use case so far, I think we can avoid creating a fully abstract function, for now, to keep the code simpler.

Reply:
Isn't this more safer to keep everything avoiding removing all data by mistake? I builded this function with this exact purpose in mind, if we add later something more than project, we would want to add/update this new property without worrying about any project property already existing inside the file.
If we want to erase (outdated or not) data from the config file, we can still pass { project: null } as data. 🤔

Reply:
[Update after discussion carried out of this thread]: we concluded that both approaches could be prone to errors (both replacing and preserving existing data). We resolved to rework the service with Create, Retrieve, Update, and Patch functions.